src/interpreter/executors/instruction/tests/reserved_registers.rs

```diff
@@ -0,0 +1,140 @@
+use super::*;
+use quickcheck::TestResult;
+use quickcheck_macros::quickcheck;
+
+// Ensure none of the opcodes can write to reserved registers
+#[quickcheck]
+fn cant_write_to_reserved_registers(opcode: u8, fuzz: u16) -> TestResult {
+    // opcode || REG_ONE || random fuzz
+    let attempt_reserved_reg_access = ((opcode as u32) << 24) | (1 << 18) | ((fuzz >> 4) as u32);
```

Comment:
It will be maybe too much coupled if we operate the internals of an instruction here. Maybe we benefit from [Instruction::from(opcode)](https://docs.rs/fuel-asm/0.7.0/fuel_asm/struct.Instruction.html#impl-From%3COpcode%3E-for-Instruction)?

Reply:
The difficulty is that there's not an easy way to programmatically generate randomized Opcode's without handling each enum variant separately. The Instruction type would be a little easier to arbitrarily muck with for fuzzing, but we'd likely need  a `test-helpers` feature to allow its internals to be set.

Reply:
Changed this test to avoid the bitmath, and now it generates instructions from a random u32 instead. This required broadening the scope of the test to consider writing to other registers besides REG_ONE.